src/pages/MatchExperiencesCatalogScreen/MatchExperiencesCatalogScreen.tsx

```diff
@@ -1,7 +1,157 @@
-// import classes from './matchExperiences-catalog-screen.module.scss';
+import { ShareMatchExperienceModal } from '@components/ShareMatchExperienceModal';
+import classes from './match-experiences-catalog-screen.module.scss';
+import { PlusOutlined } from '@ant-design/icons';
+import { FaFutbol } from 'react-icons/fa';
+import { Button, Select, Spin } from 'antd';
 
-export interface MatchExperiencesCatalogScreenProps {}
+import { useState } from 'react';
+import { MatchExperienceService } from '@/api/services/match-experience.service';
+import { MatchDetails } from '@components/MatchDetails';
+import { Screen } from '@components/Screen';
+import { useQuery } from '@tanstack/react-query';
+import { getPictureFullUrl } from '@/utils/picture.utils.ts';
+import moment from 'moment';
+import { Heart, MessageCircle } from 'lucide-react';
+import { ROUTES } from '@/constants/routes.const';
+import { useNavigate } from 'react-router-dom'; // ✅ Import for navigation
 
-export const MatchExperiencesCatalogScreen = () => {
-    return <div></div>;
+export interface MatchExperiencesCatalogScreenProps {
+    mode: 'all' | 'my';
+}
+
+const currentUserId = '67d592743f1f3c317f54c7de'; // creator
+const PageItemsLimit = 5; // Number of items per page
+
+export const MatchExperiencesCatalogScreen = ({ mode }: MatchExperiencesCatalogScreenProps) => {
+    const [page, setPage] = useState(1);
+    const [sortBy, setSortBy] = useState('date');
+
+    const navigate = useNavigate();
+
+    const {
+        data: matchExperiencesData,
+        error: matchExperiencesError,
+        isPending: matchExperiencesIsPending,
+        refetch,
+    } = useQuery({
+        queryKey:
+            mode === 'all' ? ['matchExperiences', page, sortBy] : ['userMatchExperiences', page, sortBy, currentUserId],
```

Comment:
```suggestion
            mode === 'all' ? [QUERY_KEYS.MATCH_EXPERNIENCES, page, sortBy] : [QUERY_KEYS.USER_MATCH_EXPERNIENCES, page, sortBy, currentUserId],
```

add this query keys to already existing `const QUERY_KEYS` so that it will can be used in other places as well if needed

Reply:
7251bf2706d37adb37fc56c90c28cf98858d6de9